### PR TITLE
feat(form-v2/design-1): add design drawer, enable users to edit estimated time and form instructions

### PR DESCRIPTION
**Pull request by justynoh**
## Problem
This PR implements the start page view in the form builder and enables users to edit and save their estimated time and instructions for the form start page. 

Makes progress on #3362

## Solution
**Key Changes**
- Added `DesignDrawer` and `useDesignStore`.
- Updated `FormBuilder.tsx` to display form's `startPage`.
- Additional mutations and queries added to `admin-form/common`. 

**Other Changes**
- Refactor `FormBuilder` and `BuilderFields` files (just moved some functions around for better clarity and modularity).
- Updated some `endPage` builder files (for consistency/symmetry and clarity between `startPage` and `endPage` builder code)

**Breaking Changes** 
- [X] No - this PR is backwards compatible  

**Future PRs**
- feat(form-v2/design-2): logo radio buttons and form color theme selection [#3362] 
- feat(form-v2/design-3): addition of title alignment and banner options _(more complex - will require model updates)_ [#3362]
- feat(form-v2/design-4): open design drawer when start page is clicked in builder [#4159] 

## Screenshots

image

## Tests
- [X] Open a new form and edit the design.
- [X] Open an existing form and edit the design.

### CODE REVIEW

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/BuilderFields.tsx [hunk 1]
```diff
@@ -34,3 +51,55 @@ export const BuilderFields = memo(
...
+    if (stateData.state === BuildFieldState.CreatingField) {
+      return getFormFieldsWhileCreating(existingFields, stateData)
+    }
+    return existingFields
+  }, [formData, stateData])
+
+  return {
+    builderFields,
+  }
+}
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/BuilderFields.tsx]: was there any reason why this is moved into BuilderFields.tsx? Seemed much easier to find as its own file.

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/EndPageView.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+export const EndPageView = ({ ...props }: FlexProps): JSX.Element => {
+  const { data: form } = useAdminForm()
+  const { endPageData, setState } = useEndPageBuilderStore((state) => ({
+    endPageData: endPageDataSelector(state),
+    setState: setStateSelector(state),
+  }))
+
+  useEffect(() => {
+    if (form) setState(form.endPage)
+  }, [form, setState])
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/EndPageView.tsx]: this was not in the previous code, is this needed? what bug was this fixing? (also pls refrain from adding changes unrelated to the PR haha hard to review)

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+  setStateSelector,
+  startPageDataSelector,
+  useDesignStore,
+} from '../useDesignStore'
+
+import { EmptyFormPlaceholder } from './BuilderAndDesignPlaceholder/EmptyFormPlaceholder'
+import BuilderAndDesignPlaceholder from './BuilderAndDesignPlaceholder'
+import { BuilderFields, useBuilderFields } from './BuilderFields'
+
+const FormBuilderStartPage = () => {
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx]: should we move this to its own component?

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    startPageData: startPageDataSelector(state),
+    setState: setStateSelector(state),
+  }))
+
+  useEffect(() => {
+    if (form) setState(form.startPage)
+  }, [form, setState])
+
+  // Color theme options and other design stuff, identical to public form
+
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx]: nit: extra new line

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+      startPageData.colorTheme
+        ? `theme-${startPageData.colorTheme}.500`
+        : `neutral.200`,
+    [startPageData.colorTheme],
+  )
+
+  const estTimeString = useMemo(() => {
+    if (!startPageData.estTimeTaken) return ''
+    return simplur`${startPageData.estTimeTaken} min[|s] estimated time to complete`
+  }, [startPageData.estTimeTaken])
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx]: if these are the same, any reason why we did not just reuse our current useFormHeader and useFormBannerLogo hooks? 

we can generify that hook by allowing `form` to be passed into them:

```tsx
const useFormBannerLogo = (form) => {
  ....
}

// in public form hook using the generified hook
const usePublicFormBannerLogo = () => {
  const { form } = usePublicFormContext()
  const { ...originalHookStuff } = useFormBannerLogo(form)
  ...
}

// Our admin form hook can look like:
const useBuilderFormBannerLogo = () => { 
  const { form } = useAdminFormOrTheStoreWeGotThisFrom()
  ... you get the idea 
}
```

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+
+  //
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx]: nit: leftover comment? should be deleted.

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+    }
+  }, [form, startPageData, logoBucketUrl])
+
+  const logoImgAlt = useMemo(
+    () => (form ? `Logo for ${form.admin.agency.fullName}` : undefined),
+    [form],
+  )
+
+  return (
+    <>
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx]: Should we figure out if it is worth it to also reuse the existing FormHeader component? Form builder view is supposed to reflect what the current public form looks like - will also be useful to only need to maintain a single component.

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditEndPageDrawer/EditEndPage.tsx [hunk 8]
```diff
@@ -133,5 +132,5 @@ export const EndPageBuilderInput = ({
 export const EditEndPage = (): JSX.Element => {
   const { data: form } = useAdminForm()
 
-  return <>{form ? <EndPageBuilderInput endPage={form.endPage} /> : null}</>
+  return form ? <EndPageBuilderInput endPage={form.endPage} /> : <></>
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditEndPageDrawer/EditEndPage.tsx]: nit: can probably return `null` in this case, we just need to update the return type of this component

```tsx
export const EditEndPage = (): JSX.Element | null => {
```

File: frontend/src/features/admin-form/create/builder-and-design/useDesignStore.tsx [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    startPageData: {
+      logo: { state: FormLogoState.Default },
+      colorTheme: FormColorTheme.Blue,
+    },
+    setState: (startPage) => {
+      const current = get()
+      if (isEqual(current.startPageData, startPage)) return
+      set({
+        startPageData: startPage,
+      })
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/useDesignStore.tsx]: useDesignStore seems like its functions could be in useBuilderAndDesignStore (after all it's in the name... haha)

File: frontend/src/features/admin-form/create/builder-and-design/useDesignStore.tsx [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+  setState: (startPage: FormStartPage) => void
+}
+
+export const useDesignStore = create<DesignStore>(
+  devtools((set, get) => ({
+    startPageData: {
+      logo: { state: FormLogoState.Default },
+      colorTheme: FormColorTheme.Blue,
+    },
+    setState: (startPage) => {
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/useDesignStore.tsx]: nit: the name of this function could probably be updated for clarity? how about setStartPageData?

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/EndPageView.tsx (hunk 2 above)
Comment (justynoh) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/EndPageView.tsx]: same thing as what we were talking about earlier - the old store state from a different form gets loaded first, before getting updated to the correct form state

this fixes the issue in the start page builder so i added the same to the end page builder that's all haha

File: frontend/src/features/admin-form/create/builder-and-design/useDesignStore.tsx (hunk 9 above)
Comment (justynoh) [reply frontend/src/features/admin-form/create/builder-and-design/useDesignStore.tsx]: yeah i considered it. This was a conscious decision, and I was going to rename `useBuilderAndDesignStore` later on to just `useBuilderStore` in a separate PR. Reason for this is that in `BuilderAndDesignDrawer`, we decide which drawer to show based on the `DrawerTabs` active tab.

image

Adding the start page design code to this complicates it by adding a case for a state that a user should never get to in runtime - specifically, one in which `DrawerTabs.Builder` is active, but the BuilderState is `editingStartPage`. Of course, it's the tradeoff between having a bit more code vs not being statically guaranteed. I think the tradeoff is worth it. Feel free to opine though

File: frontend/src/features/admin-form/create/builder-and-design/useDesignStore.tsx (hunk 10 above)
Comment (justynoh) [reply frontend/src/features/admin-form/create/builder-and-design/useDesignStore.tsx]: update the corresponding for endpage also? hahah

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx (hunk 6 above)
Comment (justynoh) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx]: lol oops

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx (hunk 4 above)
Comment (justynoh) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx]: oh lol i usually leave an extra line for comments relating to blocks but sure haha

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/BuilderFields.tsx (hunk 1 above)
Comment (justynoh) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/BuilderFields.tsx]: ... shoot. not sure why I did that. will revert

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditEndPageDrawer/EditEndPage.tsx (hunk 8 above)
Comment (justynoh) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditEndPageDrawer/EditEndPage.tsx]: errrr that would have other effects upstream i think? (in `BuilderAndDesignContent`)

we could do that in another smaller PR

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/EndPageView.tsx (hunk 2 above)
Comment (karrui) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/EndPageView.tsx]: feel like this is just fixing the symptom instead of the cause. like discussed, maybe we should reset the store on unmount instead of setting it on mount.

```tsx
useEffect(() => {
  return () => {
    resetStore()
  }
}, [])
```

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx (hunk 5 above)
Comment (justynoh) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx]: ok, agreed. but also, a lot of stuff in `features/admin-form` references stuff in `features/public-form` (not vice-versa though). we should probably move all of this stuff to `features/form/utils` in a separate PR?

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx (hunk 3 above)
Comment (justynoh) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx]: I thought about it but i didn't think it was big enough yet to move. 

looking forward, it'll definitely become bigger so i think its worth to pre-emptively move in this PR bah. will do!

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditEndPageDrawer/EditEndPage.tsx (hunk 8 above)
Comment (karrui) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditEndPageDrawer/EditEndPage.tsx]: won't have other effects. can you try? should work as per normal

File: frontend/src/features/admin-form/create/builder-and-design/useDesignStore.tsx (hunk 10 above)
Comment (karrui) [reply frontend/src/features/admin-form/create/builder-and-design/useDesignStore.tsx]: ye

Comment (justynoh) [reply frontend/src/features/admin-form/create/builder-and-design/useDesignStore.tsx]: okei

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditEndPageDrawer/EditEndPage.tsx (hunk 8 above)
Comment (justynoh) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditEndPageDrawer/EditEndPage.tsx]: oop you are right hahah

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/EndPageView.tsx (hunk 2 above)
Comment (justynoh) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/EndPageView.tsx]: It would make sense if we change the data fields to be possibly undefined. Can do!

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx (hunk 7 above)
Comment (justynoh) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx]: ok, managed to figure this out. Additionally, i think we should do this for the end page component as well in a separate PR. welp.

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx (hunk 5 above)
Comment (justynoh) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx]: See #4186

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx (hunk 7 above)
Comment (justynoh) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FormBuilder.tsx]: See #4187

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/StartPageView.tsx [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+export const StartPageView = () => {
+  const { data: form } = useCreateTabForm()
+  const startPageFromStore = useDesignStore(startPageDataSelector)
+
+  // Switch over to store's start page once the store is populated (done when
+  // user opens the drawer)
+  const startPage = useMemo(
+    () => (startPageFromStore ? startPageFromStore : form?.startPage),
+    [startPageFromStore, form?.startPage],
+  )
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/StartPageView.tsx:26]: same here

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/StartPageView.tsx [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+
+  return (
+    <>
+      <FormBannerLogo {...formBannerLogoProps} />
+      <FormHeader
+        title={form?.title}
+        estTimeString={estTimeString}
+        titleBg={titleBg}
+        titleColor={titleColor}
+        showHeader={true}
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/StartPageView.tsx]: nit: react props operate on presence of the prop. do you think it is cleaner to not assign `true`?
```tsx
<FormHeader
  showHeader // equivalent to showHeader={true}
  ...
/>
```

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/BuilderAndDesignContent.tsx [hunk 13]
```diff
@@ -175,20 +33,12 @@ export const BuilderAndDesignContent = ({
...
-          }
-        />
-      </Flex>
-    </>
+    <Flex flex={1} bg="neutral.200" overflow="auto">
+      {stateData.state === BuildFieldState.EditingEndPage ? (
+        <EndPageView />
+      ) : (
+        <FormBuilder placeholderProps={placeholderProps} />
+      )}
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/BuilderAndDesignContent.tsx]: should we revert this change? Hiding the form builder will make the transitions more responsive than rerendering.

File: frontend/src/features/admin-form/common/mutations.ts [hunk 14]
```diff
@@ -316,7 +317,25 @@ export const useMutateFormPage = () => {
   const queryClient = useQueryClient()
   const toast = useToast({ status: 'success', isClosable: true })
 
-  const updateFormData = useCallback(
+  const updateStartPageFormData = useCallback(
+    (newData: StartPageUpdateDto) => {
+      queryClient.setQueryData(adminFormKeys.startPage(formId), newData)
```
Comment (karrui) [inline frontend/src/features/admin-form/common/mutations.ts]: as discussed, this does nothing. we only need to update the admin form.
should remove both the start and end page keys since they are not actually used for queries

File: frontend/src/features/admin-form/common/mutations.ts [hunk 15]
```diff
@@ -334,7 +353,24 @@ export const useMutateFormPage = () => {
...
+      toastDescription: string
+    }) => {
+      toast.closeAll()
+      updateStartPageFormData(newData)
+      toast({
+        description: toastDescription,
+      })
+    },
+    [toast, updateStartPageFormData],
+  )
```
Comment (karrui) [inline frontend/src/features/admin-form/common/mutations.ts]: nit: suggest inlining since it is only used once

File: frontend/src/features/admin-form/common/queries.ts [hunk 16]
```diff
@@ -26,6 +26,7 @@ export const adminFormKeys = {
     [...adminFormKeys.id(id), 'collaborators'] as const,
   previewForm: (id: string) =>
     [...adminFormKeys.id(id), 'previewForm'] as const,
+  startPage: (id: string) => [...adminFormKeys.id(id), 'startPage'] as const,
   endPage: (id: string) => [...adminFormKeys.id(id), 'endPage'] as const,
```
Comment (karrui) [inline frontend/src/features/admin-form/common/queries.ts]: these keys should be deleted since they are not used

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/BuilderAndDesignContent.tsx (hunk 13 above)
Comment (karrui) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/BuilderAndDesignContent.tsx]: maybe should add reasoning as a comment

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/EndPageView.tsx [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+export const EndPageView = ({ ...props }: FlexProps): JSX.Element => {
+  const { data: form } = useAdminForm()
+  const endPageFromStore = useEndPageBuilderStore(endPageDataSelector)
+
+  // Switch over to store's end page once the store is populated (done when
+  // user opens customization page)
+  const endPage = useMemo(
+    () => (endPageFromStore ? endPageFromStore : form?.endPage),
+    [endPageFromStore, form?.endPage],
+  )
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/EndPageView.tsx]: should we encapsulate this process of retrieving endPage (and startPage) into a custom hook ala `useBuilderFields`?

make it more obvious the relationship between what to show

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/BuilderAndDesignDrawer.tsx [hunk 18]
```diff
@@ -47,21 +48,20 @@ export const BuilderAndDesignDrawer = (): JSX.Element | null => {
 
   const renderDrawerContent = useMemo(() => {
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/BuilderAndDesignDrawer.tsx]: maybe add return type to ensure we cover all switch cases

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/DesignDrawer/DesignDrawer.tsx [hunk 19]
```diff
@@ -0,0 +1,175 @@
...
+  resetStartPageDataSelector,
+  setStartPageDataSelector,
+  useDesignStore,
+} from '../../useDesignStore'
+import { validateNumberInput } from '../../utils/validateNumberInput'
+import { CreatePageDrawerCloseButton } from '../CreatePageDrawerCloseButton'
+import { DrawerContentContainer } from '../EditFieldDrawer/edit-fieldtype/common/DrawerContentContainer'
+
+interface DesignDrawerInputProps {
+  startPage: FormStartPage
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/DesignDrawer/DesignDrawer.tsx]: do we need the prop if this is not reused?  can just invoke the hook in the DesignDrawer component

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/DesignDrawer/DesignDrawer.tsx [hunk 20]
```diff
@@ -0,0 +1,175 @@
...
+}
+
+const DesignDrawerInput = ({
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/DesignDrawer/DesignDrawer.tsx]: nit: probably can call this DesignDrawer instead? since this component is more than one input.

File: frontend/src/features/public-form/components/FormStartPage/useFormBannerLogo.tsx [hunk 21]
```diff
@@ -0,0 +1,36 @@
+import { useMemo } from 'react'
+
+import { AdminFormDto, PublicFormDto } from '~shared/types'
+import { FormLogoState } from '~shared/types/form/form_logo'
+
+import { useEnv } from '~features/env/queries'
+
+export const useFormBannerLogo = (form?: AdminFormDto | PublicFormDto) => {
```
Comment (karrui) [inline frontend/src/features/public-form/components/FormStartPage/useFormBannerLogo.tsx]: instead of the entire form object, maybe we should just pass in the needed props in so we do not need to update the entire form on changes

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/StartPageView.tsx [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
+
+import { FormBannerLogo } from '~features/public-form/components/FormStartPage/FormBannerLogo'
+import { FormHeader } from '~features/public-form/components/FormStartPage/FormHeader'
+import { useFormBannerLogo } from '~features/public-form/components/FormStartPage/useFormBannerLogo'
+import { useFormHeader } from '~features/public-form/components/FormStartPage/useFormHeader'
+
+import { useCreateTabForm } from '../useCreateTabForm'
+import { startPageDataSelector, useDesignStore } from '../useDesignStore'
+
+export const StartPageView = () => {
```
Comment (karrui) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/StartPageView.tsx:14]: we should probably reuse public form's start page component so to maintain single source of truth, and also so we can be sure the preview is accurate

File: frontend/src/features/admin-form/common/mutations.ts [hunk 23]
```diff
@@ -313,80 +313,8 @@ export const useMutateCollaborators = () => {
...
   if (!formId) throw new Error('No formId provided')
-
-  const queryClient = useQueryClient()
   const toast = useToast({ status: 'success', isClosable: true })
 
-  const updateStartPageFormData = useCallback(
-    (newData: StartPageUpdateDto) => {
-      queryClient.setQueryData(adminFormKeys.startPage(formId), newData)
-      // Only update adminForm if it already has prior data.
-      queryClient.setQueryData<AdminFormDto | undefined>(
```
Comment (karrui) [inline frontend/src/features/admin-form/common/mutations.ts]: i think we will still need to update the query for the entire form, since this mutation will indeed update that portion

File: frontend/src/features/admin-form/common/mutations.ts [hunk 24]
```diff
@@ -313,80 +313,8 @@ export const useMutateCollaborators = () => {
...
-      )
-    },
-    [formId, queryClient],
-  )
-
-  const updateEndPageFormData = useCallback(
-    (newData: EndPageUpdateDto) => {
-      queryClient.setQueryData(adminFormKeys.endPage(formId), newData)
-      // Only update adminForm if it already has prior data.
-      queryClient.setQueryData<AdminFormDto | undefined>(
```
Comment (karrui) [inline frontend/src/features/admin-form/common/mutations.ts:323]: same here.

File: frontend/src/features/public-form/components/FormStartPage/FormHeader.tsx [hunk 25]
```diff
@@ -1,98 +1,101 @@
...
+              variant="solid"
+              colorScheme="primary"
+              aria-label="Mobile section sidebar"
+              fontSize="1.5rem"
+              icon={<BxMenuAltLeft />}
+              d={{ base: 'flex', md: 'none' }}
+              onClick={onMobileDrawerOpen}
+            />
+          ) : (
+            <></>
```
Comment (karrui) [inline frontend/src/features/public-form/components/FormStartPage/FormHeader.tsx]: nit: not changed by your PR, but should be change this to `null` instead?

File: frontend/src/features/public-form/components/FormStartPage/useFormBannerLogo.tsx [hunk 26]
```diff
@@ -1,35 +1,45 @@
...
-import { useEnv } from '~features/env/queries'
-
-export const useFormBannerLogo = (form?: AdminFormDto | PublicFormDto) => {
-  const { data: { logoBucketUrl } = {} } = useEnv(
-    form?.startPage.logo.state === FormLogoState.Custom,
-  )
+interface UseFormBannerLogoInputs {
+  logoBucketUrl?: string
+  logo?: FormLogo
+  agency?: AgencyBase
```
Comment (karrui) [inline frontend/src/features/public-form/components/FormStartPage/useFormBannerLogo.tsx:9]: nit: use AgencyDto instead?

File: frontend/src/features/admin-form/common/mutations.ts (hunk 23 above)
Comment (justynoh) [reply frontend/src/features/admin-form/common/mutations.ts]: oh yes, we should still do that

File: frontend/src/features/public-form/components/FormStartPage/useFormBannerLogo.tsx (hunk 26 above)
Comment (justynoh) [reply frontend/src/features/public-form/components/FormStartPage/useFormBannerLogo.tsx:9]: the type of `form.admin.agency` is `AgencyBase` though, and none of the extra info is needed from `AgencyDto`. is there a reason for using `AgencyDto` instead?

Comment (karrui) [reply frontend/src/features/public-form/components/FormStartPage/useFormBannerLogo.tsx:9]: because that's the real field that is being used, but i guess that's fine too.

**Review by karrui — APPROVED** [review]
lgtm
